main/lsp/test/error_reporter_test.cc

```diff
@@ -0,0 +1,191 @@
+// has to go first as it violates our requirements
+#include "common/kvstore/KeyValueStore.h"
+#include "core/Error.h"
+#include "core/ErrorQueue.h"
+#include "core/Loc.h"
+#include "core/Unfreeze.h"
+#include "main/lsp/ErrorReporter.h"
+#include "main/lsp/LSPMessage.h"
+#include "main/lsp/LSPOutput.h"
+#include "main/lsp/json_types.h"
+#include "payload/payload.h"
+#include "spdlog/sinks/null_sink.h"
+#include "test/helpers/MockFileSystem.h"
+using namespace std;
+
+namespace sorbet::realmain::lsp::test {
+namespace {
+
+options::Options makeOptions(string_view rootPath) {
+    options::Options opts;
+    opts.rawInputDirNames.emplace_back(string(rootPath));
+    opts.runLSP = true;
+    opts.fs = make_shared<sorbet::test::MockFileSystem>(rootPath);
+    return opts;
+}
+auto nullSink = make_shared<spdlog::sinks::null_sink_mt>();
+auto nullOpts = makeOptions("");
+auto logger = make_shared<spdlog::logger>("console", nullSink);
+
+shared_ptr<LSPConfiguration> makeConfig() {
+    auto config = make_shared<LSPConfiguration>(nullOpts, make_shared<LSPOutputToVector>(), logger, true, false);
+    InitializeParams initParams("", make_unique<ClientCapabilities>());
+    initParams.rootPath = "";
+    initParams.initializationOptions = make_unique<SorbetInitializationOptions>();
+    config->setClientConfig(make_shared<LSPClientConfiguration>(initParams));
+    return config;
+}
+
+unique_ptr<core::GlobalState> makeGS() {
+    auto gs = make_unique<core::GlobalState>((make_shared<core::ErrorQueue>(*logger, *logger)));
+    unique_ptr<const OwnedKeyValueStore> kvstore;
+    payload::createInitialGlobalState(gs, nullOpts, kvstore);
+    gs->errorQueue->ignoreFlushes = true;
+    return gs;
+}
+
+} // namespace
+TEST(ErrorReporterTest, NotifiesVSCodeWhenFileHasErrors) {
+    auto gs = makeGS();
+    auto cs = makeConfig();
+    ErrorReporter er(cs);
+    vector<unique_ptr<core::Error>> errors;
+    {
+        core::UnfreezeFileTable fileTableAccess(*gs);
+        auto epoch = 0;
+        auto file = make_shared<core::File>("foo.rb", "foo", core::File::Type::Normal, epoch);
+        auto fref = gs->enterFile(file);
+        auto outputVector = dynamic_pointer_cast<LSPOutputToVector>(cs->output);
+
+        errors.emplace_back(
+            make_unique<core::Error>(core::Loc(fref, 0, 0), core::ErrorClass{1, core::StrictLevel::True}, "MyError",
+                                     vector<core::ErrorSection>(), vector<core::AutocorrectSuggestion>(), false));
+        er.pushDiagnostics(epoch, fref, errors, *gs);
+
+        auto output = outputVector->getOutput();
+        EXPECT_EQ(1, output.size());
+
+        auto &message = output.back();
+        auto &notificationMessage = message->asNotification();
+        auto &publishDiagnosticParams = get<unique_ptr<PublishDiagnosticsParams>>(notificationMessage.params);
+
+        EXPECT_EQ(publishDiagnosticParams->uri, cs->fileRef2Uri(*gs, fref))
+            << fmt::format("Reports file with errors to VS code");
```

Comment:
(Nit) You're not actually using any formatting strings, so this can be simplified.

```suggestion
            << "Reports file with errors to VS code";
```